src/zsf_config.c

```diff
@@ -20,8 +19,15 @@ int _zsf_ini_handler(char *section, char *key, char *value, void *data_ptr) {
 
   errno = 0;
   if (!strcmp(section, "sealock")) {
-    config_ptr->num_locks = 1;
-    if (!strcmp(key, "computation_mode")) {
+    sealock_index_t lock_index = config_ptr->num_locks - 1;
+    assert(!*key || (*key && lock_index >= 0));
+    if (!*key) {
+      if (config_ptr->num_locks < ZSF_MAX_LOCKS) {
+        config_ptr->num_locks++;
+      }
```

Comment:
Should there be an error or log message if we exceed ZSF_MAX_LOCKS?

Reply:
The function now returns with an INI_FAIL if we run out of sea lock slots.